assets/src/stories-editor/components/media-inserter/index.js

```diff
@@ -114,56 +145,14 @@ const mediaPicker = ( dialogTitle, mediaType, updateBlock ) => {
 };
 
 const applyWithSelect = withSelect( ( select ) => {
-	const { getCurrentPage } = select( 'amp/story' );
-	const { getBlock } = select( 'core/block-editor' );
-	const { isReordering, getSettings } = select( 'amp/story' );
-
-	const currentPage = getCurrentPage();
-	const block = getBlock( currentPage );
-	const mediaType = block && block.attributes.mediaType ? block.attributes.mediaType : '';
-	const { allowedVideoMimeTypes } = getSettings();
+	const { isReordering } = select( 'amp/story' );
 
 	return {
 		isReordering: isReordering(),
-		canInsertBlockType: ( name ) => isBlockAllowedOnPage( name, currentPage ),
-		// As used in <HeaderToolbar> component
-		showInserter: select( 'core/edit-post' ).getEditorMode() === 'visual' && select( 'core/editor' ).getEditorSettings().richEditingEnabled,
-		mediaType,
-		allowedVideoMimeTypes,
-	};
-} );
-
-const applyWithDispatch = withDispatch( ( dispatch, props, { select } ) => {
-	const { getCurrentPage } = select( 'amp/story' );
-	const { getBlockOrder } = select( 'core/block-editor' );
-	const { insertBlock } = dispatch( 'core/block-editor' );
-
-	return {
-		insertBlock: ( name ) => {
-			const currentPage = getCurrentPage();
-			const index = getBlockOrder( currentPage ).length;
-
-			const insertedBlock = createBlock( name, {} );
-
-			insertBlock( insertedBlock, index, currentPage );
-		},
-		updateBlock: ( media ) => {
-			const clientId = getCurrentPage();
-			const { updateBlockAttributes, selectBlock } = dispatch( 'core/block-editor' );
-
-			if ( ! clientId ) {
-				return;
-			}
-
-			const processed = processMedia( media );
-			updateBlockAttributes( clientId, processed );
-			selectBlock( clientId );
-		},
 	};
 } );
 
 export default compose(
 	applyWithSelect,
-	applyWithDispatch,
 	ifCondition( ( { isReordering } ) => ! isReordering ),
```

Comment:
What's this weird thing doing here? Can we get rid of it and then get rid of the whole `compose` thing? What even is this?

Reply:
`ifCondition` is a HOC that only renders the component if the given condition is satisfied. Basically a shortcut for doing `if ( isReordering ) { return null; }` in the component.

Reply:
https://github.com/WordPress/gutenberg/blob/f68a107c777169cd80b3ce56ec4f7c8360f2178c/packages/compose/src/higher-order/if-condition/README.md

Reply:
And yes, we can get rid of it